### ISSUE TITLE
Batch Inference

### ISSUE THREAD

**Issue by filipetrocadoferreira** [facebookresearch/Detectron#84]
How to easily to take advantage of batch processing during inference?

**Comment by kampelmuehler** [comment]
Any news on this? Is supporting batches for testing planned for the near future?

**Comment by kampelmuehler** [comment]
@filipetrocadoferreira I've implemented this for Faster RCNN and so far it seems to work ok. It's not a whole lot of work to do this yourself.

**Comment by filipetrocadoferreira** [comment]
Which files did you change?

**Comment by kampelmuehler** [comment]
All changes are to be made in `lib/core/test.py` and `lib/core/test_engine.py`. (+ adding a TEST.IMS_PER_BATCH config option)

**Comment by filipetrocadoferreira** [comment]
Nice, feel free to post the commit :+1:

**Comment by kampelmuehler** [comment]
Sure, there you go:
kampelmuehler/Detectron@e3cb235067e4c0a476ea5113edc108bcc975a107
note that it's not a very clean implementation.

**Comment by filipetrocadoferreira** [comment]
Thanks a lot :)
Do you think it needs changes for Mask/Keypoint prediction?

**Comment by kampelmuehler** [comment]
Yes it will for sure. See whereever it says `Not implemented`, or better compare with the non `_batch` methods.

**Comment by filipetrocadoferreira** [comment]
Keypoint Batch Inference is not straight forward. I'm facing some problems
